llvm/docs/InstCombineContributorGuide.md

```diff
@@ -554,3 +554,11 @@ guidelines.
    use of ValueTracking queries. Whether this makes sense depends on the case,
    but it's usually a good idea to only handle the constant pattern first, and
    then generalize later if it seems useful.
+
+## Guidelines for reviewers
+
+ * Do not ask contributors to implement non-splat vector support in code
```

Comment:
maybe this should be "new reviewers". IIUC the motivation is that its complicated and may be too much for a novice.

Reply:
Done. The problem is not just that it can be complicated, but that it is likely to significantly increase the number of required review iterations or even result in directly contradictory review feedback. We don't have a strong consensus on what exactly is acceptable when implementing non-splat vector support or what the best way of going about it is.

Reply:
Hopefully #85676 will change that :)